### PR TITLE
feat: update react peer dependency to react@18

### PR DESCRIPTION
**Pull request by kabaros**
Implements [LIBS-703](https://dhis2.atlassian.net/browse/LIBS-703)

---

### Description

In preparation for bumping react version in app-platform, we need to bump the peer dependency of the UI - it works without it but generates a lot of unnecessary warnings.

Tested locally by ensuring `yarn why react` only shows react@18 in  UI, and by adding the library through `yalc`, there were no warnings about the peer dependencies anymore in `app-scripts@alpha` (I had the peer dependencies as `>=16.13` first, and that didn't work, but changing it to `^16.13 || ^18` worked)

[LIBS-703]: https://dhis2.atlassian.net/browse/LIBS-703?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,8 +55,8 @@
         "@babel/parser": "^7.17.0",
         "@badeball/cypress-cucumber-preprocessor": "^20.0.2",
         "@cypress/webpack-preprocessor": "^6.0.1",
-        "@dhis2/app-runtime": "^3.9.0",
-        "@dhis2/cli-app-scripts": "^11.7.0",
+        "@dhis2/app-runtime": "^3.11.2",
+        "@dhis2/cli-app-scripts": "^12.0.0-alpha.19",
```
Comment (kabaros) [inline package.json:59]: not sure how to manage this - probably should keep it as the version in master, until it's merged and bump it to v12

**Review by Mohammer5 — APPROVED** [review]
LGTM

---

* [x] Storybook works locally
* [x] `yarn install && yarn setup && yarn build` works locally, even in a newly cloned repository
* [x] Tests pass

**Review by amcgee — APPROVED** [review]

**Comment by kabaros** [comment]
@Mohammer5 @amcgee I am re-requesting review because I did some significant changes [here](https://github.com/dhis2/ui/pull/1624/commits/48ad26c7ccb6ddfed7813a9abace7a13f9b80e81) - the tests were failing after the upgrade, part of it could be solved by using the enzyme's unofficial adapter for React 18, but I thought we should bump testing-library as well since we're doing this .. this trickled to a lot of changes related to userEvent becoming async and the different way hooks are tested now. 

All these changes fall under the ones we documented for v12 migration guide [here](https://github.com/dhis2/app-platform/blob/19ba4ab83660b3bc0d9d13396bf766972397df94/docs/migration/v12.md#react-testing-library).

**Comment by kabaros** [comment]
sorry folks, this was on auto-merge :/

**Pull request merged**